src/shared/application/use-cases/__tests__/UpdateTaskUseCase.test.ts

```
@@ -57,8 +58,19 @@ describe("UpdateTaskUseCase", () => {
 
     // Mock transaction to execute the callback immediately
     vi.mocked(mockDatabase.transaction).mockImplementation(
-      async (mode, tables, callback) => {
-        return await callback();
+      (mode, tables, callback) => {
+        const result = callback({} as any);
+        const mockPromise = {
+          then: (onFulfilled?: any, onRejected?: any) =>
```

Biome (2.1.2)

[error] Do not add then to an object.



(lint/suspicious/noThenProperty)